game-core/src/main/java/tools/image/PolygonGrabber.java

```diff
@@ -113,525 +113,537 @@ public static void main(final String[] args) {
+    private void guessCountryName(final JTextField text, final Iterable<Entry<String, Point>> centersiter) {
+      final List<String> options = new ArrayList<>();
+      for (final Entry<String, Point> item : centersiter) {
+        final Point p = new Point(item.getValue());
+        for (final Polygon polygon : current) {
+          if (polygon.contains(p)) {
+            options.add(item.getKey());
+          } // if
+        } // while
       } // while
-    } // while
-    if (!options.isEmpty()) {
-      Collections.shuffle(options);
-      text.setText(options.get(0));
+      if (!options.isEmpty()) {
+        Collections.shuffle(options);
+        text.setText(options.get(0));
+      }
     }
-  }
 
-  /**
-   * Checks to see if the given point is of color black.
-   */
-  private boolean isBlack(final Point p) {
-    return isBlack(p.x, p.y);
-  }
-
-  /**
-   * Checks to see if the x/y coordinates from a given point
-   * are inbounds and if so is it black.
-   */
-  private boolean isBlack(final int x, final int y) {
-    if (!inBounds(x, y)) {
-      // not inbounds, can't be black
-      return false;
+    /**
+     * Checks to see if the given point is of color black.
+     */
+    private boolean isBlack(final Point p) {
+      return isBlack(p.x, p.y);
     }
-    // gets ARGB integer value and we LOGICAL AND mask it
-    // with ARGB value of 00,FF,FF,FF to determine if it
-    // it black or not.
-    // maybe here ?
-    return (bufferedImage.getRGB(x, y) & 0x00FFFFFF) == 0;
-  }
 
-  private static boolean isBlack(final int x, final int y, final BufferedImage bufferedImage) {
-    if (!inBounds(x, y, bufferedImage)) {
-      // not inbounds, can't be black
-      return false;
+    /**
+     * Checks to see if the x/y coordinates from a given point
+     * are inbounds and if so is it black.
+     */
+    private boolean isBlack(final int x, final int y) {
+      if (!inBounds(x, y)) {
+        // not inbounds, can't be black
+        return false;
+      }
+      // gets ARGB integer value and we LOGICAL AND mask it
+      // with ARGB value of 00,FF,FF,FF to determine if it
+      // it black or not.
+      // maybe here ?
+      return (bufferedImage.getRGB(x, y) & 0x00FFFFFF) == 0;
     }
-    // gets ARGB integer value and we LOGICAL AND mask it
-    // with ARGB value of 00,FF,FF,FF to determine if it
-    // it black or not.
-    // maybe here ?
-    return (bufferedImage.getRGB(x, y) & 0x00FFFFFF) == 0;
-  }
 
-  /**
-   * Checks if the given x/y coordinate point is inbounds or not.
-   */
-  private boolean inBounds(final int x, final int y) {
-    return x >= 0 && x < bufferedImage.getWidth(null) && y >= 0 && y < bufferedImage.getHeight(null);
-  }
-
-  private static boolean inBounds(final int x, final int y, final Image image) {
-    return x >= 0 && x < image.getWidth(null) && y >= 0 && y < image.getHeight(null);
-  }
+    private boolean isBlack(final int x, final int y, final BufferedImage bufferedImage) {
```

Comment:
It looks like 'static' was dropped from the signature of this method. Was that optional, or mandatory and I'm missing something?

Reply:
Compiler: "The method inBounds cannot be declared static; static methods can only be declared in a static or top level type"

Reply:
Ah okay, interesting; non-static inner class; it makes sense now why static was dropped - thanks!